packages/editor/src/lib/editor/Editor.ts

```diff
@@ -290,6 +290,11 @@ export class Editor extends EventEmitter<TLEventMap> {
 			if (startShape && endShape) {
 				// if arrow has two bindings, always parent arrow to closest common ancestor of the bindings
 				nextParentId = this.findCommonAncestor([startShape, endShape]) ?? parentPageId
+
+				// if arrow has two bindings to the same shape, parrent arrow to it
+				if (startShape.id === endShape.id) {
+					nextParentId = startShape.id
```

Comment:
This would parent the arrow to any shape, not just frames. I agree with the basic concept of this PR (and am surprised that I haven't noticed it before!) but we'd have to achieve it in a slightly different way.